hackathon_site/hardware/api_filters.py

```diff
@@ -0,0 +1,24 @@
+from django_filters import rest_framework as filters
+from .models import Hardware
+from .serializers import HardwareSerializer
+from django_filters.fields import Lookup
+
+
+class HardwareFilter(filters.FilterSet):
+    queryset = Hardware
+    serializer_class = HardwareSerializer
+
+    class Meta:
+        model = Hardware
+        fields = ["name", "quantity_available"]
+
+    def comma_separate_filter(self, queryset, value):
+        value_list = value.split(",")
+        queryset = super().filter(queryset, Lookup(value_list, "in"))
+
+        return queryset
+
+    in_stock = filters.BooleanFilter(
+        field_name="quantity_remaining", lookup_expr="gt", exclude=False
+    )
+    category_ID = filters.CharFilter(method="comma_separate_filter")
```

Comment:
OK after more digging, it seems there is actually a CSV filter that's built in called `BaseInFilter` (I guess they expect you to extend it, but it works on its own).

You can change this line to:
`ids = filters.BaseInFilter(field_name="ids")`

If you ctrl+click your way into the source code, it basically just sets the lookup expr to `in`, and then passes in the comma separated list you get as-is, which is legal sql apparently (I would have thought it needs brackets, but this works fine in my testing)

Reply:
Note: this is the build-in csv filter. 

![image](https://user-images.githubusercontent.com/36619969/125322668-9a547280-e370-11eb-9be4-388371d3d96c.png)